### ISSUE TITLE
Music player: more playback options

### ISSUE THREAD

**Issue by tobiasKaminsky** [nextcloud/android#1038]
Wished via mail:
During audio playback show an option to play the whole folder.
For saving data plan this should be limited to already downloaded files.
This can be done like in VLC:
- repeat current
- play whole directory
- loop directory
- shuffle

**Comment by aronovgj** [comment]
>  For saving data plan this should be limited to already downloaded files.

Why not let the user choose?

**Comment by tobiasKaminsky** [comment]
I thought this would clutter the playback UI.
And if you really want to have all files in a folder, you can easily click on this folder and choose "synchronize".

But I am not the design expert: @jancborchardt @eppfel

**Comment by GitHubGeek** [comment]
Workaround: Use a music playback app that support custom music files locations, and then point it to `Android/data/com.nextcloud.client/files/...` 😄

**Comment by eppfel** [comment]
I consider this an edge-case. Nextcoud should not become a music player app. The discussion about sync and ux shows how this complicates things.

Preview to "check" a file seems enough...

**Comment by jancborchardt** [comment]
Yeah – while music files are also important, we should focus on images and videos which are a much bigger use-case. :)

**Comment by LukeOwncloud** [comment]
I like the @tobiasKaminsky's idea about enhancing the music player just a little bit. However, I also understand that there are other more important things to do. As compromise I suggest to let the user decide which music player to use, i.e.: Do not open the internal music player as default. Make the open action an Intent instead, allowing the user to choose which app to use applying standard Android means.

I know, option "open with" is accessible via menu, but it is cumbersome to reach (long press, menu, choose action). @GitHubGeek, don't you agree? And @jancborchardt, you already agreed that using intents would be nice 2 years ago: https://github.com/owncloud/android/issues/775 👍

**Comment by LukeOwncloud** [comment]
Related issue https://github.com/nextcloud/android/issues/914

**Comment by trummerschlunk** [comment]
I agree, that the internal player should not become a full featured player, but some basic options would be very much appreciated. I am a mastering engineer and I check masters by listening on the phone while taking a walk. Having to initiate playback for every tracks is tiring and gets me out of the album flow every time.
Thanks for your time and great work!

**Comment by yasuoiwakura** [comment]
hi there,
i got feedback from our non-techie users who listen to mp3-lectures:

**when finished playing an mp3-file, the player should start the next file.**

imho that's not really an "advanced" feature but just what you expect when plugging in your USB or CD into your radio: Just play it till the end.

ofc there are better setups for "listening" like cloudBeats, ampache, Subsonic etc. but the nextcloud client login is just as simple as you wish for "non-techie"-users.
Plus, they **don't even want** fancy player-apps with confusing buttons.

**Comment by ezaquarii** [comment]
@tobiasKaminsky @AndyScherzinger  Although our player infrasrtructure already supports the playlist concep alreadyt, we do not ship any UI experience on top of it.

So in principles this should be possible. However, it's a non-trivial UX effort to do so properly.

**Comment by ezaquarii** [comment]
@yasuoiwakura I personally just sync my music files and use 3rd party music player to play this media forlder.

**Comment by yasuoiwakura** [comment]
> @yasuoiwakura I personally just sync my music files and use 3rd party music player to play this media forlder.

Hi there,
i discovered cloudbeats as a convenient nextcloud-client for listening.

Nevertheless, For the first easiest setup I recommend the nextcloud client (especially for non-techy users) so you don't have to fiddle around but just use the cloud.

**Comment by AndyScherzinger** [comment]
Agreeing with @ezaquarii and like mentioned also think that implementing (proper) playlist support is a rather huge effort ((pre-)fetching audio files)

**Comment by yasuoiwakura** [comment]
> Agreeing with @ezaquarii and like mentioned also think that implementing (proper) playlist support is a rather huge effort ((pre-)fetching audio files)

Hi there,
Would it also be a big effort to automatically choose and play the alphabetically next file when finishing one file?

**Comment by AndyScherzinger** [comment]
The answer would be yes, since:
You can't just play the next file since it might not be available on your device, so you need to fetch it first also it would need to be defined _when_ to fetch the next file. The same would apply for a playlist.

So while from a process/step description it seems easy it still takes quite some effort to implement this.

**Comment by ezaquarii** [comment]
We support streaming from server and the player operates on OCFile abstraction so can handle that.

Overall UX is of primary concern tho. What we don't support is file caching. Current player impl would re-stream the file on each play, which can easily eat mobile data from your monthly allowance.

**Comment by TechupBusiness** [comment]
I would also try to use third-party player for most media functionality (ideally VLC). I personally miss a "Play folder" function, which only exists for single files. Or at least an info to easily find the local files in the nextcloud application folder without reading docs first.
